### PR TITLE
Jormungandr: POC of a SIRI lite connector for realtime

### PR DESCRIPTION
**Pull request by kinnou02**
This connector is based on the STIF implementation of StopMonitoring from SIRI 2.0 as defined by
the local agreement IDF 2.4
(http://www.normes-donnees-tc.org/wp-content/uploads/2014/05/Profil_Siri_IDF_V2-4-STIF-20130712.pdf)

### CODE REVIEW

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    """
+    class managing calls to a siri lite external service providing real-time next passages
+    """
+    def __init__(self, id, service_url, object_id_tag=None, timeout=10, **kwargs):
+        self.service_url = service_url
+        self.timeout = timeout  # timeout in seconds
+        self.rt_system_id = id
+        self.object_id_tag = object_id_tag if object_id_tag else id
+        self.breaker = pybreaker.CircuitBreaker(fail_max=app.config.get('CIRCUIT_BREAKER_MAX_SIRILITE_FAIL', 5),
+                                                reset_timeout=app.config.get('CIRCUIT_BREAKER_CLEVERAGE_TIMEOUT_S', 60))
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: CIRCUIT_BREAKER_SIRILITE_TIMEOUT_S

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        line_id = route_point.fetch_line_id(self.object_id_tag)
+
+        if not stop_id or not line_id:
+            # one a the id is missing, we'll not find any realtime
+            logging.getLogger(__name__).debug('missing realtime id for {obj}: stop code={s}'.
+                                              format(obj=route_point, s=stop_id))
+            self.record_internal_failure('missing id')
+            return None
+
+        url = "{base_url}{stop_id}".format(base_url=self.service_url, stop_id=stop_id)
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/realtime_schedule/siri_lite.py:93]: so the line_id is not used ?

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        return url
+
+    def _get_dt(self, datetime_str):
+        return pytz.utc.localize(datetime.strptime(datetime_str, "%Y-%m-%dT%H:%M:%S.%fZ"))
+
+    def _get_passages(self, route_point, resp):
+        logging.getLogger(__name__).debug('sirilite response: {}'.format(resp))
+
+        line_code = route_point.fetch_line_id(self.object_id_tag)
+        monitored_stop_visit = resp['siri']['serviceDelivery']['stopMonitoringDelivery']['monitoredStopVisit']
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: are the fields mandatory ? shouldn't you use `.get` ?

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        schedules = next((vj for vj in monitored_stop_visit if vj['monitoredVehicleJourney']['lineRef']['value'] == line_code), None)
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: schedules doesn't seems used

**Review by antoine-de — COMMENTED** [review]
great work!

It would be nice to add test though (with mocks to have examples of for the siri response)

**Comment by antoine-de** [comment]
so do you think it will be possible to filter to only have the route's passage ?

**Comment by kinnou02** [comment]
there is no route in the response, and even so there isn't a shared referential of route(that is required to have meaningful id) , we could use the destination to find, then

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py (hunk 2 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/realtime_schedule/siri_lite.py:93]: actually we can't use the `lineRef` parameter, but we filter manually the response

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py (hunk 3 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: what did you not understand in "POC"? :D

Comment (antoine-de) [reply source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: :mask:

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py (hunk 2 above)
Comment (antoine-de) [reply source/jormungandr/jormungandr/realtime_schedule/siri_lite.py:93]: ok can you thus add a comment there?

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    def _make_url(self, route_point):
+        """
+        The url returns something like a departure on a stop point
+        """
+
+        stop_id = route_point.fetch_stop_id(self.object_id_tag)
+        line_id = route_point.fetch_line_id(self.object_id_tag)
+
+        if not stop_id or not line_id:
+            # one a the id is missing, we'll not find any realtime
```
Comment (pbougue) [inline source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: `if` the id is missing ?

File: source/jormungandr/jormungandr/realtime_schedule/siri_lite.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        return url
+
+    def _get_dt(self, datetime_str):
+        return pytz.utc.localize(datetime.strptime(datetime_str, "%Y-%m-%dT%H:%M:%S.%fZ"))
+
+    def _get_passages(self, route_point, resp):
+        logging.getLogger(__name__).debug('sirilite response: {}'.format(resp))
+
+        line_code = route_point.fetch_line_id(self.object_id_tag)
+        monitored_stop_visit = resp.get('siri', {}).get('serviceDelivery', {}).get('stopMonitoringDelivery', {}).get('monitoredStopVisit', [])
```
Comment (pbougue) [inline source/jormungandr/jormungandr/realtime_schedule/siri_lite.py]: line length

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**